#!/usr/bin/env python
PACKAGE = "travesim_adapters"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Multicast range 224.0.0.0 through 239.255.255.255. See:
# https://www.iana.org/assignments/multicast-addresses/multicast-addresses.xhtml
gen.add("multicast_address", str_t, 0, "Multicast Address",  "224.0.0.1")

# User Ports (1024-49151). See:
# https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml
gen.add("multicast_port", int_t, 0, "Multicast Port", 10002,  1024, 49151)

gen.add("reset_vision", bool_t, 0, "Reset vision adapter",  False)

exit(gen.generate(PACKAGE, "vision_adapter", "Vision"))
